apollo-router/src/plugins/connectors/plugin.rs

```diff
@@ -0,0 +1,263 @@
+
+        if debug_extensions {
+            tracing::warn!(
+                "Connector debugging is enabled, this may expose sensitive information."
+            );
+        }
+
+        Ok(Connectors { debug_extensions })
+    }
+
+    fn supergraph_service(&self, service: supergraph::BoxService) -> supergraph::BoxService {
+        let conf_enabled = self.debug_extensions;
+        service
+            .map_future_with_request_data(
+                move |req: &supergraph::Request| {
+                    let is_enabled = conf_enabled
+                        && req
+                            .supergraph_request
+                            .headers()
+                            .get(CONNECTORS_DEBUG_HEADER_NAME)
+                            == Some(&HeaderValue::from_static("true"));
+                    if is_enabled {
+                        req.context.extensions().with_lock(|mut lock| {
+                            lock.insert::<ConnectorContext>(ConnectorContext::new());
+                        });
+                    }
+
+                    is_enabled
+                },
+                move |is_enabled: bool, f| async move {
+                    let mut res: supergraph::ServiceResult = f.await;
+
+                    res = match res {
+                        Ok(mut res) => {
+                            if is_enabled {
+                                if let Some(debug) = res
+                                    .context
+                                    .extensions()
+                                    .with_lock(|mut lock| lock.remove::<ConnectorContext>())
+                                {
+                                    let (parts, stream) = res.response.into_parts();
+                                    let (mut first, rest) = stream.into_future().await;
+
+                                    if let Some(first) = &mut first {
+                                        first.extensions.insert(
+                                            "apolloConnectorsDebugging",
+                                            json!({"version": "1", "data": debug.serialize() }),
+                                        );
+                                    }
+                                    res.response = http::Response::from_parts(
+                                        parts,
+                                        once(ready(first.unwrap_or_default())).chain(rest).boxed(),
+                                    );
+                                }
+                            }
+
+                            Ok(res)
+                        }
+                        Err(err) => Err(err),
+                    };
+
+                    res
+                },
+            )
+            .boxed()
+    }
+}
+
+register_plugin!("apollo", "preview_connectors", Connectors);
+
+// === Structs for collecting debugging information ============================
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
Suggest replacing the `new` constructor by just adding `Default` here
```suggestion
#[derive(Debug, Clone, Serialize, Deserialize, Default)]
```